### PR TITLE
DBZ-1781 Add support for sharded MongoDB collections

### PR DESCRIPTION
**Pull request by jgraf50**

### CODE REVIEW

File: debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/RecordMakers.java [hunk 1]
```diff
@@ -195,11 +196,11 @@ public int recordEvent(Document oplogEvent, Document masterEvent, Instant timest
             final Map<String, ?> offset = source.lastOffset(replicaSetName);
             Document patchObj = oplogEvent.get("o", Document.class);
             // Updates have an 'o2' field, since the updated object in 'o' might not have the ObjectID ...
-            Object o2 = oplogEvent.get("o2");
-            String objId = o2 != null ? idObjToJson(o2) : idObjToJson(patchObj);
+            Document queryObj = oplogEvent.get("o2", Document.class);
```
Comment (gunnarmorling) [inline debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/RecordMakers.java]: So is this guaranteed to always be a document, also if there's just a single id field / no shard key?

Comment (jgraf50) [reply debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/RecordMakers.java]: Yes if it is not null then it is a Mongo document.  You can see a simple example here (https://www.compose.com/articles/the-mongodb-oplog-and-node-js/) and I have verified on our own database that non-sharded collection updates just look like `o2: {"_id": "foobar"}`.

Comment (gunnarmorling) [reply debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/RecordMakers.java]: Ok, thanks!

File: debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/RecordMakers.java [hunk 2]
```diff
@@ -195,11 +196,11 @@ public int recordEvent(Document oplogEvent, Document masterEvent, Instant timest
...
+            String objId = queryObj != null ? idObjToJson(queryObj) : idObjToJson(patchObj);
```
Comment (gunnarmorling) [inline debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/RecordMakers.java]: As `idObjToJson()` only ever receives `Document` now, I'll change its signature/implementation accordingly.

**Comment by gunnarmorling** [comment]
Rebased and applied. Thanks a lot, @jgraf50!

**Pull request closed without merging**